### PR TITLE
Extend Liquid Drops model to support config_yaml and options

### PR DESCRIPTION
**Pull request by kwkwan**

### CODE REVIEW

File: lib/expressir/express/express_remarks_decorator.rb [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+      end
+
+      def initialize(remark, options)
+        @remark = remark
+        @options = options
+      end
+
+      def call
+        result = remark
+        if options["leveloffset"]
```
Comment (ronaldtse) [inline lib/expressir/express/express_remarks_decorator.rb:19]: Let’s delete this option which is not needed.

File: lib/expressir/liquid/declarations/schema_drop.rb [hunk 2]
```diff
@@ -6,16 +6,52 @@ module Declarations
       class SchemaDrop < ::Expressir::Liquid::DeclarationDrop
         include ::Expressir::Liquid::IdentifierDrop
 
-        def initialize(model)
+        def initialize(model, selected_schemas: nil, options: {}) # rubocop:disable Lint/MissingSuper
```
Comment (ronaldtse) [inline lib/expressir/liquid/declarations/schema_drop.rb:9]: Why do we need the notion of “selected schemas” in Expressir? Can’t we do it in liquid?

File: lib/expressir/liquid/model_element_drop.rb [hunk 3]
```diff
@@ -3,8 +3,10 @@
 module Expressir
   module Liquid
     class ModelElementDrop < ::Liquid::Drop
-      def initialize(model) # rubocop:disable Lint/MissingSuper
+      def initialize(model, selected_schemas: nil, options: {}) # rubocop:disable Lint/MissingSuper
```
Comment (ronaldtse) [inline lib/expressir/liquid/model_element_drop.rb:6]: The identifier does not need any option?

File: lib/expressir/liquid/repository_drop.rb [hunk 4]
```diff
@@ -5,16 +5,22 @@
...
+        @options = options
         super
       end
 
       def schemas
         return [] unless @model.schemas
 
         @model.schemas.map do |item|
-          ::Expressir::Liquid::Declarations::SchemaDrop.new(item)
+          ::Expressir::Liquid::Declarations::SchemaDrop.new(
```
Comment (ronaldtse) [inline lib/expressir/liquid/repository_drop.rb:19]: You could make a new method RepositoryDrop.filter that returns the selected SchemaDrop objects.

File: lib/expressir/model/model_element.rb [hunk 5]
```diff
@@ -151,10 +151,10 @@ def to_hash(root_path: nil, formatter: nil, include_empty: nil, select_proc: nil
       end
 
       # @return [Liquid::Drop]
-      def to_liquid
```
Comment (ronaldtse) [inline lib/expressir/model/model_element.rb:154]: We should not change this.

File: lib/expressir/model/model_element.rb [hunk 6]
```diff
@@ -151,10 +151,10 @@ def to_hash(root_path: nil, formatter: nil, include_empty: nil, select_proc: nil
...
+      def to_liquid(selected_schemas: nil, options: nil)
         klass_name = "#{self.class.name.gsub('::Model::', '::Liquid::')}Drop"
         klass = Object.const_get(klass_name)
-        klass.new(self)
+        klass.new(self, selected_schemas: selected_schemas, options: options)
```
Comment (ronaldtse) [inline lib/expressir/model/model_element.rb:157]: Only need to update RepositoryDrop, if it cannot be done in Liquid…

File: lib/expressir/express/express_remarks_decorator.rb (hunk 1 above)
Comment (kwkwan) [reply lib/expressir/express/express_remarks_decorator.rb:19]: The option `leveloffset`  has been removed via 676ca05ed3d461301912ad035875a806bcdad94d

**Pull request merged**